### PR TITLE
Add no gravity link support

### PR DESCRIPTION
**Pull request by AzulRadio**
# 🎉 New feature

Original issue: gazebosim/gz-sim#504

## Summary

Add parsing and support for no gravity link when using DartSim engine

For implementation and test details, see gazebosim/gz-sim#2398

Changes in other repo:
gz-physics: gazebosim/gz-physics#633
sdformat: gazebosim/sdformat#1410

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: dartsim/src/SDFFeatures.cc [hunk 1]
```diff
@@ -600,6 +600,11 @@ Identity SDFFeatures::ConstructSdfLink(
 
   bodyProperties.mInertia.setLocalCOM(localCom);
 
+  // ignmsg << "SDF gravity status = " << _sdfLink.EnableGravity() << std::endl;
```
Comment (ahcorde) [inline dartsim/src/SDFFeatures.cc:603]: remove

File: dartsim/src/SDFFeatures.cc [hunk 2]
```diff
@@ -600,6 +600,11 @@ Identity SDFFeatures::ConstructSdfLink(
...
+
+  bodyProperties.mGravityMode = _sdfLink.EnableGravity();
+  // ignmsg << "gravity in dart set to: " << bodyProperties.mGravityMode << std::endl;
```
Comment (ahcorde) [inline dartsim/src/SDFFeatures.cc:606]: remove

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by ahcorde — APPROVED** [review]
we need to wait to merge the PR in sdformat https://github.com/gazebosim/sdformat/blob/sdf14/python/test/pyLink_TEST.py

**Comment by iche033** [comment]
@osrf-jenkins run tests please

File: dartsim/src/WorldFeatures_TEST.cc [hunk 3]
```diff
@@ -190,6 +196,12 @@ TEST_F(WorldFeaturesFixture, Gravity)
     EXPECT_PRED_FORMAT2(vectorPredicate3,
                         Eigen::Vector3d(0.5, 0, 2.5),
                         pos);
+    // pose for link without gravity should not change
+    Eigen::Vector3d posNoGravity = linkNoGravity->FrameDataRelativeToWorld()
+                        .pose.translation();
+    EXPECT_PRED_FORMAT2(vectorPredicate3,
+                        Eigen::Vector3d(10, 10, 10),
+                        pos);
```
Comment (iche033) [inline dartsim/src/WorldFeatures_TEST.cc:204]: CI caught a test failure here:

```
  /github/workspace/dartsim/src/WorldFeatures_TEST.cc:202: Failure
  Eigen::Vector3d(10, 10, 10) and pos ([10 10 10] and 
[0.50050000000579775                   0  2.5005000000132527]) are not equal
```

Comment (iche033) [reply dartsim/src/WorldFeatures_TEST.cc:204]: should `pos` be changed to `posNoGravity`?

Comment (AzulRadio) [reply dartsim/src/WorldFeatures_TEST.cc:204]: I think you are right.

Comment (iche033) [reply dartsim/src/WorldFeatures_TEST.cc]: updated in [5f4c033](https://github.com/gazebosim/gz-physics/pull/633/commits/5f4c03345b07f5b29a3be134026fde0045dc4c63)

**Review by iche033 — APPROVED** [review]

**Pull request merged**